### PR TITLE
Clean up profiles

### LINKED ISSUE #82: Remember last used values for 'custom'
1. Change values so filter displays `custom`
2. Select a saved filter
3. Select `custom` again

Current behavior: No change in the active filter
Desired behavior: Filter changes back to what it was in step 1

Then you can make `custom` selectable from the notification, so that you can test out a certain configuration and compare it to existing filters, without continually saving/deleting filters or complicating the app interface.

### PR DESCRIPTION
**Pull request by smichel17**
Closes #82, #75, #84

- [x] Add per filter lower brightnes
- [x] Make the default profiles the same as any other profile, as a pre-requisite
- [x] Add a way to reset profiles to the default state.
- [x] Make the `Custom` profile persistent
- [x] Add graceful profile format update so people don't lose their saved profiles.

### CODE REVIEW

**Comment by smichel17** [comment]
@raatmarien Current master, 7e1a772, is ready to be released as 2.10.2. Whenever you get the chance to do that, this branch, 16d0f78, is ready to be merged and can be released as 3.10.0.

Even though this is ready, I would release 2.10.2 also, in case people need to downgrade, since 2.10.2 should be pretty stable but 3.0.0 hasn't been tested as thoroughly.

**Comment by smichel17** [comment]
Hmm. I'm not sure how much time I will have to fix bugs that will probably crop up, so it might make sense to hold off on this release for a little.

**Comment by smichel17** [comment]
@raatmarien Note that I am rebasing this off of master and force pushing to it. I'll let you know each time I do that.

(Actually, I have a `rebaser` branch that's at 4d81308; I merge master into that (let's call this commit M), then `reset --hard` it to 1a95d4f, then merge master again and resolve the merge by `git checkout M -- *`. Then checkout profiles++ and rebase off the rebaser branch)

**Comment by smichel17** [comment]
force pushed.

**Comment by smichel17** [comment]
force pushed. With 502a4f9, this is probably *actually* stable enough to release.

File: app/src/main/java/com/jmstudios/redmoon/util/Util.kt [hunk 1]
```diff
@@ -36,116 +36,18 @@
...
-            upgradeFrom(28)
-        } else -> {
-            Log.e("handleUpgrades", "Didn't catch upgrades from version $version")
-            upgradeFrom(version+1)
-        }
-    }
-
-    upgradeFrom(if (Config.introShown) Config.fromVersionCode else BuildConfig.VERSION_CODE)
-}
+fun getColor (resId: Int): Int = ContextCompat.getColor(appContext, resId)
```
Comment (smichel17) [inline app/src/main/java/com/jmstudios/redmoon/util/Util.kt]: Note the global getColor and getString functions. Very nice and short.

File: app/src/main/java/com/jmstudios/redmoon/util/Util.kt [hunk 2]
```diff
@@ -36,116 +36,18 @@
  */
 package com.jmstudios.redmoon.util
```
Comment (smichel17) [inline app/src/main/java/com/jmstudios/redmoon/util/Util.kt]: Util.kt is now for univeral helper functions. It's small enough that I'll usually import it with a wildcard into every class.

File: app/src/main/java/com/jmstudios/redmoon/view/ScreenFilterView.kt [hunk 3]
```diff
@@ -14,281 +14,24 @@
...
-import android.graphics.Color
 import android.view.View
+import com.jmstudios.redmoon.helper.Logger
 
-import com.jmstudios.redmoon.preference.ColorSeekBarPreference
-import com.jmstudios.redmoon.preference.DimSeekBarPreference
-import com.jmstudios.redmoon.preference.IntensitySeekBarPreference
+import com.jmstudios.redmoon.helper.Profile
 
 class ScreenFilterView(context: Context) : View(context) {
```
Comment (smichel17) [inline app/src/main/java/com/jmstudios/redmoon/view/ScreenFilterView.kt]: Like the presenter, this is just removing stuff.

File: app/src/main/java/com/jmstudios/redmoon/view/ScreenFilterView.kt [hunk 4]
```diff
@@ -14,281 +14,24 @@
...
- *     in all copies.
- *
- *     THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL
- *     WARRANTIES WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED
- *     WARRANTIES OF MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE
- *     AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT, INDIRECT, OR
- *     CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS
- *     OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT,
- *     NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
- *     CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
```
Comment (smichel17) [inline app/src/main/java/com/jmstudios/redmoon/view/ScreenFilterView.kt]: There's so little left in this class, I removed the derivative copyright notice (it's added in the places this was moved to).

File: app/src/main/res/layout/preference_profile_selector.xml [hunk 5]
```diff
@@ -33,7 +33,7 @@
       android:id="@+id/profile_action_button"
       android:layout_width="wrap_content"
       android:layout_height="wrap_content"
-      android:text="@string/button_add_profile"
+      android:text="@string/button_add_filter"
```
Comment (smichel17) [inline app/src/main/res/layout/preference_profile_selector.xml]: None (or, very few) of the changes to the xml files are functional. Just take a look at `res/values/strings.xml` and you'll see the changes and why it's so many lines of code. Nice and consistent now, though :)

File: build.gradle [hunk 6]
```diff
@@ -6,7 +6,7 @@ buildscript {
         jcenter()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:2.2.3'
+        classpath 'com.android.tools.build:gradle:2.3.0'
```
Comment (smichel17) [inline build.gradle]: gradle 2.3.0 is a significant performance improvement.

**Comment by raatmarien** [comment]
This is great, a huge improvement, especially for the code quality. Well done!

Does this mean master is ready for v3.0.0 release or are there more things you think should be done before that?

**Pull request merged**